js/ui/context_menu/ui.menu_base.js

```diff
@@ -177,18 +180,44 @@ class MenuBase extends HierarchicalCollectionWidget {
         this._initActions();
     }
 
+    _getLinkContainer(iconContainer, textContainer, { linkAttr, url }) {
+        if(iconContainer) {
+            iconContainer.addClass(DX_ICON_WITH_URL_CLASS);
+        }
+        if(textContainer) {
+            textContainer.addClass(DX_MENU_ITEM_CAPTION_URL_CLASS);
+        }
+        const linkAttributes = isObject(linkAttr) ? linkAttr : {};
+        return $('<a>')
+            .addClass(DX_ITEM_URL_CLASS)
+            .attr({ ...linkAttributes, href: url })
+            .append(iconContainer)
+            .append(textContainer);
+    }
+
+    _addContent($container, itemData) {
+        const { html, url } = itemData;
+
+        const iconContainer = this._getIconContainer(itemData);
+        const textContainer = this._getTextContainer(itemData);
+
+        $container.html(html);
+        if(url) {
+            const link = this._getLinkContainer(iconContainer, textContainer, itemData);
+            $container.append(link);
+        } else {
+            $container
+                .append(iconContainer)
+                .append(textContainer);
+        }
+        $container.append(this._getPopoutContainer(itemData));
+        this._addContentClasses(itemData, $container.parent());
+    }
+
     _getTextContainer(itemData) {
-        const { text, url, linkAttr } = itemData;
+        const { text } = itemData;
         const $itemContainer = $('<span>').addClass(DX_MENU_ITEM_CAPTION_CLASS);
         const itemContent = isPlainObject(itemData) ? text : String(itemData);
```

Comment:
Now there can be a different type of content in item, so let's name it more specific
```suggestion
        const itemText = isPlainObject(itemData) ? text : String(itemData);
```